frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -446,6 +446,105 @@ void main(List<String> args) async {
+    expect(
+        () async => await CustomStruct.staticReturnCustomStructError(bridge: api), throwsA(isA<CustomStructError>()));
+  });
+
+  test('Do not throw CustomStructError static method', () async {
+    expect(await CustomStruct.staticReturnCustomStructOk(bridge: api), 3);
+  });
+
+  test('Throw CustomNestedError1', () async {
+    expect(() async => await api.returnCustomNestedError1(),
+        throwsA(CustomNestedError1.errorNested(CustomNestedError2.customNested2Number(3))));
+  });
+
+  test('Throw CustomNestedError1 variant 1', () async {
+    expect(
+        () async => await api.returnCustomNestedError1Variant1(), throwsA(CustomNestedError1.customNested1("custom")));
+  });
+
+  test('Throw CustomNestedError2', () async {
+    expect(() async => await api.returnCustomNestedError2(), throwsA(CustomNestedError2.customNested2("custom")));
+  });
+
+  test('Throw CustomError variant 0', () async {
+    expect(() async => await api.returnErrorVariant(variant: 0), throwsA(isA<CustomError>()));
+  });
+
+  test('Throw CustomError variant 1', () async {
+    expect(() async => await api.returnErrorVariant(variant: 1), throwsA(isA<CustomError>()));
+  });
+
+  test('Do not throw CustomError', () async {
+    expect(await api.returnOkCustomError(), 3);
+  });
+
+  test('Throw CustomError static method', () async {
+    expect(() async => await SomeStruct.staticReturnErrCustomError(bridge: api), throwsA(isA<CustomError>()));
+  });
+
+  test('Throw CustomError static method, verifies implements Frb', () async {
+    expect(() async => await SomeStruct.staticReturnErrCustomError(bridge: api), throwsA(isA<FrbException>()));
+  });
+
+  test('Do not throw CustomError static method', () async {
+    expect(await SomeStruct.staticReturnOkCustomError(bridge: api), 3);
+  });
+
+  test('Do not throw CustomError', () async {
+    expect(await api.returnOkCustomError(), 3);
+  });
+
+  test('Throw CustomError nonstatic method', () async {
+    expect(() async => await SomeStruct(bridge: api, value: 7).nonStaticReturnErrCustomError(),
+        throwsA(isA<CustomError>()));
+    bool didCatch = false;
+    try {
+      await SomeStruct(bridge: api, value: 7).nonStaticReturnErrCustomError();
+    } catch (e) {
+      final FrbBacktracedException ex = e as FrbBacktracedException;
+      print("backtrace: ${ex.backtrace}");
+      assert(ex.backtrace.contains("wire_non_static_return_err_custom_error__method__SomeStruct::"));
+      didCatch = true;
+    }
+    assert(didCatch);
+  });
+
+  test('Do not throw CustomError nonstatic method', () async {
+    expect(await SomeStruct(bridge: api, value: 6).nonStaticReturnOkCustomError(), 6);
+  });
+
+  test('Throw anyhow error', () async {
+    expect(() async => await api.throwAnyhow(), throwsA(isA<FrbException>()));
+  });
+
+  test('Function with custom result panics', () async {
+    expect(() async => await api.panicWithCustomResult(), throwsA(isA<FrbException>()));
```

Comment:
Maybe we should also assert the *content* of FrbException. For example, some stack traces are inside. If it is hard to assert the whole stack trace string, at least we can check the stack trace string *contains* some fragments.